### PR TITLE
<WIP> eclipse/deeplearning4j#8483 Generate information on ignored tests

### PR DESCRIPTION
**Pull request by eraly**
Works fine. Just ugly. Might redo not in bash..Or comment to improve readability

### CODE REVIEW

**Comment by eraly** [comment]
I was redoing the title to get it to link to the issue properly...fail.

File: java-testing/ignores/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+For dl4j:
+```
+./summarize-ignores deeplearning4j results
+Ignored methods count: 6 results/deeplearning4j/ignored_classes.csv
+Ignored methods information  written to results/deeplearning4j/ignored_methods.csv
+Ignored classes count: 22 results/deeplearning4j/ignored_methods.csv
+Ignored classes information written to results/deeplearning4j/ignored_methods.csv
+```
+ ignored_classes.csv, contains the list of test classes with a @Ignore for the whole class with other pertinent information. 
+ ignored_methods.csv, contains the list of test classes with a @Ignore for the whole class with other pertinent information.
```
Comment (AlexDBlack) [inline java-testing/ignores/README.md]: copy/paste issue (second line should reference methods not class)

**Comment by eraly** [comment]
Cleaned up the scripts. Documentation remaining.

File: java-testing/ignores/parse-for-ignores [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+# Main method
+# Arguments:
+#  1) directory to run checks in relative to dl4j repo, checks files under
+#			src/test
+###############################################################################
+main() {
+  declare PROJECT_DIR="$1"
+  if [[ -z $LIBND4J_HOME ]]; then
+    echo -n "ERROR: LIBND4J_HOME is either unset or set to a non-empty string."
+    echo -n "Set LIBND4J_HOME and rerun.Exiting script..." 1>&2
```
Comment (AlexDBlack) [inline java-testing/ignores/parse-for-ignores]: Let's not use LIBND4J_HOME to locate that - we don't generally use this variable anymore, most people won't have it set.
Instead, let's assume deeplearning4j and dl4j-dev-tools are in the same parent directory
For example, deeplearning4j can be found at `../../deeplearning4j/deeplearning4j`

**Review by AlexDBlack — CHANGES_REQUESTED** [review]
Looks great, other than a couple of minor flagged issues.

Also, can we also track this sort of ignore? (this sort is in ND4J only)
`OpValidationSuite.ignoreFailing()`
https://github.com/eclipse/deeplearning4j/search?q=OpValidationSuite.ignoreFailing&unscoped_q=OpValidationSuite.ignoreFailing

**Comment by eraly** [comment]
Should be trivial to add something for OpValidationSuite.ignoreFailing(). Will do.

File: java-testing/ignores/parse-for-ignores (hunk 2 above)
Comment (eraly) [reply java-testing/ignores/parse-for-ignores]: k, will fix

File: java-testing/ignores/README.md (hunk 1 above)
Comment (eraly) [reply java-testing/ignores/README.md]: yikes..

**Review by agibsonccc — APPROVED** [review]

**Review by agibsonccc — COMMENTED** [review]
Maybe allow specifying the dl4j root directory with a command line parameter?
The rest alex covered.

File: java-testing/ignores/parse-for-ignores (hunk 2 above)
Comment (kkoj) [inline java-testing/ignores/parse-for-ignores]: It is better for the output without "-n" option.
It should be a new line in each output.

or use "-e" with "\n" to feed the line.

File: java-testing/ignores/parse-for-ignores [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    DL4J_BASE_DIR=$(dirname "$LIBND4J_HOME")
+    echo -e "\nINFO:Found DL4J base dir: $DL4J_BASE_DIR"
+    test_path="${DL4J_BASE_DIR}/${PROJECT_DIR}"
+    echo -e "\nINFO:Running checks on: ${test_path}\n"
+  fi
+  
+  if [[ ! -d "$test_path" ]]; then
+    echo -n "ERROR: $test_path DOES NOT exist. "
+    echo -n "Supplied paths have to be relative to the deeplearning4j base dir"
+    echo -n ", i.e the parent dir of libnd4j,nd4j etc Exiting script..." 1>&2
```
Comment (kkoj) [inline java-testing/ignores/parse-for-ignores]: It is better for the output without "-n" option.
It should be a new line in each output.

or use "-e" with "\n" to feed the line.

File: java-testing/ignores/parse-for-ignores [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    exit 8
+  fi
+
+  parse_test_dirs
+  echo -e "\n\nINFO: Run completed."
+}
+
+###############################################################################
+#Call to main
+main "$@"
```
Comment (kkoj) [inline java-testing/ignores/parse-for-ignores]: It is better to use the actual function for main function name, not "main".
parse_for_ignores() is the suitable name for this function.
Line 103 will be parse_for_ignore().

**Review by kkoj — COMMENTED** [review]
The file name should be parse-for-ignores.sh with .sh.

File: java-testing/ignores/parse-for-ignores (hunk 2 above)
Comment (eraly) [reply java-testing/ignores/parse-for-ignores]: I don't want a new line that is why I set -n. And I want the file to be only 80 characters.

File: java-testing/ignores/parse-for-ignores (hunk 4 above)
Comment (eraly) [reply java-testing/ignores/parse-for-ignores]: Will do.

**Comment by eraly** [comment]
Thanks for the reviews @agibsonccc @kkoj

File: java-testing/ignores/parse-for-ignores (hunk 2 above)
Comment (eraly) [reply java-testing/ignores/parse-for-ignores]: I don't want a new line that is why I do "-n". I want all that to print on one line but I want to keep the length of characters in the file to only 80 characters.

File: java-testing/ignores/parse-for-ignores (hunk 4 above)
Comment (eraly) [reply java-testing/ignores/parse-for-ignores]: Done

File: java-testing/ignores/parse-for-ignores (hunk 2 above)
Comment (eraly) [reply java-testing/ignores/parse-for-ignores]: Does that and added an optional argument.

**Comment by eraly** [comment]
Need to add  support for OpValidationSuite.ignoreFailing() and test it.

**Comment by eraly** [comment]
Added support for OpValidationSuite.ignoreFailing()  and OpValidationSuite.IGNORE_FAILING

**Comment by eraly** [comment]
k, I am ready to merge.

**Review by AlexDBlack — APPROVED** [review]
Thanks!

**Pull request merged**